backend/src/apiserver/server/api_converter_test.go

```diff
@@ -2306,3 +2314,2356 @@ func TestToApiRecurringRun(t *testing.T) {
+				UpdateTimeInSec: 100,
+				State:           model.RuntimeStateUnspecified,
+				Error:           nil,
+			},
+			false,
+			"",
+		},
+		{
+			"State",
+			&apiv2beta1.RuntimeStatus{
+				State: apiv2beta1.RuntimeState_CANCELING,
+			},
+			&model.RuntimeStatus{
+				UpdateTimeInSec: 0,
+				State:           model.RuntimeStateCancelling,
+				Error:           nil,
+			},
+			false,
+			"",
+		},
+		{
+			"Full spec",
+			&apiv2beta1.RuntimeStatus{
+				UpdateTime: &timestamppb.Timestamp{Seconds: 100},
+				State:      apiv2beta1.RuntimeState_CANCELING,
+				Error:      util.ToRpcStatus(util.NewInvalidInputError("Invalid input: %s", "sample value")),
+			},
+			&model.RuntimeStatus{
+				UpdateTimeInSec: 100,
+				State:           model.RuntimeStateCancelling,
+				Error:           util.ToError(util.ToRpcStatus(util.NewInvalidInputError("Invalid input: %s", "sample value"))),
+			},
+			false,
+			"",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := toModelRuntimeStatus(tt.apiStatus)
+			if tt.wantErr {
+				assert.NotNil(t, err)
+				assert.Contains(t, err.Error(), tt.errMsg)
+				assert.Nil(t, got)
+			} else {
+				assert.Nil(t, err)
+				assert.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func Test_toModelRuntimeStatuses(t *testing.T) {
+	arg := []*apiv2beta1.RuntimeStatus{
+		{},
+		nil,
+		{
+			Error: util.ToRpcStatus(util.NewInvalidInputError("Invalid input: %s", "sample value")),
+		},
+		{
+			UpdateTime: &timestamppb.Timestamp{Seconds: 100},
+		},
+		{
+			State: apiv2beta1.RuntimeState_CANCELING,
+		},
+		{
+			UpdateTime: &timestamppb.Timestamp{Seconds: 100},
+			State:      apiv2beta1.RuntimeState_CANCELING,
+			Error:      util.ToRpcStatus(util.NewInvalidInputError("Invalid input: %s", "sample value")),
+		},
+	}
+	expected := []*model.RuntimeStatus{
+		{
+			UpdateTimeInSec: 0,
+			State:           model.RuntimeStateUnspecified,
+			Error:           nil,
+		},
+		{
+			UpdateTimeInSec: 0,
+			State:           "",
+			Error:           nil,
+		},
+		{
+			UpdateTimeInSec: 0,
+			State:           model.RuntimeStateUnspecified,
+			Error:           util.ToError(util.ToRpcStatus(util.NewInvalidInputError("Invalid input: %s", "sample value"))),
+		},
+		{
+			UpdateTimeInSec: 100,
+			State:           model.RuntimeStateUnspecified,
+			Error:           nil,
+		},
+		{
+			UpdateTimeInSec: 0,
+			State:           model.RuntimeStateCancelling,
+			Error:           nil,
+		},
+		{
+			UpdateTimeInSec: 100,
+			State:           model.RuntimeStateCancelling,
+			Error:           util.ToError(util.ToRpcStatus(util.NewInvalidInputError("Invalid input: %s", "sample value"))),
+		},
+	}
+	got, err := toModelRuntimeStatuses(arg)
+	assert.Nil(t, err)
+	assert.Equal(t, expected, got)
+}
+
+func Test_toApiRuntimeStatus(t *testing.T) {
+	tests := []struct {
+		name        string
+		modelStatus *model.RuntimeStatus
+		want        *apiv2beta1.RuntimeStatus
+	}{
+		{
+			"nil",
+			nil,
```

Comment:
Is `nil` a valid state? Does it result to some error at a later point?

Reply:
Correct. It will throw an error at a later time. Generally, `toApi*` functions return `nil` when they encounter errors. The calling functions then check if the returned pointer is `nil` and returns an error.